### PR TITLE
bundle create: fix overwrite existing bundle bug

### LINKED ISSUE #2879: operator-sdk bundle create --generate-only is removing the CSV and CRD's content
## Bug Report

**What did you do?**
Run the following command for Memcached Go Sample upgraded to use SDK 0.17

```
 $ operator-sdk bundle create --generate-only
INFO[0000] Building annotations.yaml                    
INFO[0000] Generating output manifests directory        
INFO[0000] Building Dockerfile                          
```

**What did you expect to see?**
The bundle be created successfully

**What did you see instead? Under which circumstances?**
The command is removing the content of the CSV in the manifest dir. See:

` $ cat deploy/olm-catalog/memcached-operator/manifests/memcached-operator.clusterserviceversion.yaml
`

**Environment**
* operator-sdk version: 0.17
* go version: 1.3

**Additional context**
By @estroz :

> shows that the issue is if values of --output-dir == --directory, the library overwrites the output directory and we set --ouput-dir to --directory by default to help migrate users to the new manifests/metadata format

### LINKED ISSUE #2880: operator-sdk bundle link a sdk doc instead of OCP enhancement
## Feature Request

When we run `operator-sdk bundle` as follows it is showing a link for a doc. I think we should properly doc the commands and its purpose as address the users to the OLM docs to check its details. So, in POV it should show a link for SDK docs instead of the OCP enhancement or just add the info required in the command helper. 

```
 $ operator-sdk bundle 
Generate operator bundle metadata and build operator bundle images, which
are used to manage operators in the Operator Lifecycle Manager.

More information on operator bundle images and metadata:
https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md#docker

Usage:
  operator-sdk bundle [command]

Available Commands:
  create      Create an operator bundle image
  validate    Validate an operator bundle image

Flags:
  -h, --help   help for bundle

Global Flags:
      --verbose   Enable verbose logging

Use "operator-sdk bundle [command] --help" for more information about a command.

```

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**  print deprecation notice for old bundle format and req uest migration

**Motivation for the change:** Closes #2879 since the output directory is no longer always set. Also closes #2880, which was just a link update in command docs.

/kind bug

### CODE REVIEW

File: changelog/fragments/2879.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      bundle create: fix overwrite existing bundle bug
+
+      https://github.com/operator-framework/operator-sdk/issues/2879
```
Comment (camilamacedo86) [inline changelog/fragments/2879.yaml:7]: We do not need add the PR link any more. Could you please remove?

File: changelog/fragments/2879.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      bundle create: fix overwrite existing bundle bug
```
Comment (camilamacedo86) [inline changelog/fragments/2879.yaml:5]: WDYT?
```suggestion
      fix overwrite existing bundle bug when the command `operator-sdk bundle create --generate-only` was called.
```

File: changelog/fragments/2879.yaml (hunk 1 above)
Comment (estroz) [reply changelog/fragments/2879.yaml:7]: This is a link to the issue itself for more context.

Comment (camilamacedo86) [reply changelog/fragments/2879.yaml:7]: It will be added automatically because this we do not need to add it anymore.

**Comment by estroz** [comment]
This bug is a little more complicated than I realized. If `bundle create` does not set the output directory and `--directory` is not pointing to `deploy/olm-catalog/<operator-name>/manifests`, the file generator will not set the first `COPY` arg reproducibly.

File: changelog/fragments/2879.yaml (hunk 1 above)
Comment (estroz) [reply changelog/fragments/2879.yaml]: The pull request link will be added, not the issue.

**Comment by estroz** [comment]
Instead of adding complexity to auto-migrate users to the new bundle format, which would still be error prone, I chose the simpler approach of printing a deprecation notice if `--directory`'s value is not a `manifests/` directory with directions on how to migrate. Printing the notice still allows users to run:
```console
$ operator-sdk bundle create <--generate-only|image-tag:v0.1.0> --directory deploy/olm-catalog/test-operator/0.1.0
```
so this is a backwards compatible change.

The old bundle format is already deprecated so this PR doesn't need a deprecation changelog entry.

File: cmd/operator-sdk/bundle/cmd.go [hunk 3]
```diff
@@ -32,12 +32,17 @@ type bundleCmd struct {
 func NewCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "bundle",
-		Short: "Work with operator bundle metadata and bundle images",
-		Long: `Generate operator bundle metadata and build operator bundle images, which
-are used to manage operators in the Operator Lifecycle Manager.
+		Short: "Manage operator bundle metadata",
+		Long: `Manage bundle builds, bundle metadata generation, and bundle validation.
```
Comment (camilamacedo86) [inline cmd/operator-sdk/bundle/cmd.go:36]:
```suggestion
		Long: ` 
Manage bundle builds, bundle metadata generation, and bundle validation.
```

File: cmd/operator-sdk/bundle/create.go [hunk 4]
```diff
@@ -51,57 +52,51 @@ bundle image containing operator metadata and manifests, tagged with the
...
+https://github.com/operator-framework/operator-registry#manifest-format
 
 NOTE: bundle images are not runnable.
 `,
 		Example: `The following invocation will build a test-operator 0.1.0 bundle image using Docker.
 This image will contain manifests for package channels 'stable' and 'beta':
 
   $ operator-sdk bundle create quay.io/example/test-operator:v0.1.0 \
-      --directory ./deploy/olm-catalog/test-operator/0.1.0 \
+      --directory ./deploy/olm-catalog/test-operator/manifests \
```
Comment (camilamacedo86) [inline cmd/operator-sdk/bundle/create.go:85]: we should not use the dir manifest. It will cause error.

**Comment by estroz** [comment]
/hold

needs some discussion.

**Comment by estroz** [comment]
Closing in favor of #2944

**Pull request closed without merging**